algorithms/src/std_algorithms/Kokkos_ExclusiveScan.hpp

```diff
@@ -23,38 +23,46 @@
 namespace Kokkos {
 namespace Experimental {
 
+//
+// overload set accepting execution space
+//
+
 // overload set 1
 template <class ExecutionSpace, class InputIteratorType,
           class OutputIteratorType, class ValueType>
 std::enable_if_t<::Kokkos::Experimental::Impl::are_iterators<
-                     InputIteratorType, OutputIteratorType>::value,
+                     InputIteratorType, OutputIteratorType>::value&& ::Kokkos::
+                     is_execution_space<ExecutionSpace>::value,
                  OutputIteratorType>
 exclusive_scan(const ExecutionSpace& ex, InputIteratorType first,
                InputIteratorType last, OutputIteratorType first_dest,
                ValueType init_value) {
   static_assert(std::is_move_constructible<ValueType>::value,
                 "ValueType must be move constructible.");
-  return Impl::exclusive_scan_default_op_impl(
+  return Impl::exclusive_scan_default_op_exespace_impl(
```

Comment:
While you assert that `ValueType` must be move constructible, `init_value` is passed to `exclusive_scan_default_op_exespace_impl` by copy constructing it.

(This kind of thing happens all over)

Reply:
I think i address this in https://github.com/kokkos/kokkos/pull/6350/commits/138f4854bb222596661297650186c73876dae40d